guidance/zero123_utils.py

```diff
@@ -115,23 +119,50 @@ def train_step(self, embeddings, pred_rgb, polar, azimuth, radius, guidance_scal
 
         t = torch.randint(self.min_step, self.max_step + 1, [1], dtype=torch.long, device=self.device)
 
+        # Set weights acc to closeness in phi
+        if len(embeddings['phis']) > 1:
+            phi_deltas = [abs(azimuth.item() + embeddings['phis'][0] - phi) for phi in embeddings['phis']]
```

Comment:
I think it would be better to calculate the weights according to the cosine of the orientation (and clamp it to [0,1]) instead of based on azimuth. Doing it on azimuth alone will not work, e.g. if we have a top, bottom and front views, that all have the same azimuth value of 0.